straxen/contexts.py

```diff
@@ -261,52 +281,78 @@ def xenonnt_simulation(output_folder='./strax_data',
+                        ('cmt_run_id', cmt_run_id_sim, *cmt_options['gain_model'])})
+    fax_config_override_from_cmt = dict()
+    for fax_field, cmt_field in _config_overlap.items():
+        fax_config_override_from_cmt[fax_field] = ('cmt_run_id', cmt_run_id_sim,
+                                                   *cmt_options[cmt_field])
+    st.set_config({'fax_config_override_from_cmt': fax_config_override_from_cmt})
+
+    # and all other parameters for processing
     for option in cmt_options:
-        st.config[option] = tuple(['cmt_run_id', _cmt_run_id_proc_only, *cmt_options[option]])
+        st.config[option] = ('cmt_run_id', cmt_run_id_proc, *cmt_options[option])
 
+    # Done with "default" usage, now to overwrites from file
+    #
     # Take fax config and put into context option
-    if overwrite_from_fax_file:
+    if overwrite_from_fax_file_proc or overwrite_from_fax_file_sim:
         fax_config = straxen.get_resource(fax_config, fmt='json')
-
         for fax_field, cmt_field in _config_overlap.items():
-            st.config[cmt_field] = tuple([cmt_options[cmt_field][0] + '_constant',
-                                          fax_config[fax_field]])
-
-    # Setup simulation
-    # Pass the CMT options to simulation (override all other config input methods)
-    else:
-        fax_config_override_from_cmt = dict()
-
-        for fax_field, cmt_field in _config_overlap.items():
-            fax_config_override_from_cmt[fax_field] = tuple(['cmt_run_id', cmt_run_id, *cmt_options[cmt_field]])
-        st.set_config({'fax_config_override_from_cmt': fax_config_override_from_cmt})
-
-    # Fix gain model
-    st.set_config({'gain_model_mc': tuple(['cmt_run_id', cmt_run_id, *cmt_options['gain_model']])})
-
-    # User customized overwrite
-    cmt_options = straxen.get_corrections.get_cmt_options(st)  # This includes gain_model_mc
-    for option in cmt_option_overwrite:
+            if overwrite_from_fax_file_proc:
+                st.config[cmt_field] = ( cmt_options[cmt_field][0] + '_constant',
+                                         fax_config[fax_field])
+            if overwrite_from_fax_file_sim:
+                st.config['fax_config_override_from_cmt'][fax_field] = (
+                         cmt_options[cmt_field][0] + '_constant',fax_config[fax_field])
+
+    # And as the last step - manual overrrides, since they have the highest priority
+    # User customized for simulation
+    for option in cmt_option_overwrite_sim:
         if option not in cmt_options:
             raise ValueError(f'Overwrite option {option} is not using CMT by default '
                              'you should just use set config')
+        if not option in _config_overlap.values():
+            raise ValueError(f'Overwrite option {option} does not have mapping from '
+                              'CMT to fax config! ')
+        for fax_key,cmt_key in _config_overlap.items():
+            if cmt_key==option:
+                continue
```

Comment:
```suggestion
            if cmt_key!=option:
                continue
```
Should be `!=` is not it?

Reply:
actually, it's correct, i force it to stop iterating once it finds cmt_key equal to desired option, so i can use value of fax_key.

Reply:
But continue means you go to the next `fax_key, cmt_key` pair if your cmt_key is identical with `option`. If I get you are answer right it is exactly the opposite of what you would like to do.

Reply:
good catch! my caffeine deprived (or oversaturated) brain decided to confuse `break` with `continue`

Reply:
there is one more change in that part, i think now it should be fine.